TC intensity work

kieranricardo opened the pull request:
@wcarthur this PR contains all my TC intensity work so far - sorry its so long!

kieranricardo committed:
added refactored emanuel code

kieranricardo committed:
create and test python bindings for tc intensity model

kieranricardo committed:
windspeed exploration

kieranricardo committed:
documentation

kieranricardo committed:
load era5 data

Commit:
era5 extract working

Commit:
bran extraction

Commit:
still data quality issues

Commit:
still data quality issues

Commit:
still data quality issues

Commit:
bran script

Commit:
bran script

Commit:
edits

Commit:
edits

Commit:
edits

kieranricardo committed:
update intensity model to output rmax and vmax

Commit:
code works but reinitializing causes some errors

Commit:
code works but reinitializing causes some errors

Commit:
model working

Commit:
model working

Commit:
fixed time stepping

Commit:
extract relative humidity

kieranricardo committed:
fixed intialization issue

kieranricardo committed:
fixed intialization issue

kieranricardo committed:
ocean mixing

kieranricardo committed:
ocean mixing

kieranricardo committed:
ocean mixing kinda working - fix scaling

kieranricardo committed:
ocean mixing working correctly

kieranricardo committed:
ocean mixing working correctly

kieranricardo committed:
temp profile extract

Commit:
extract ocean temperature profile

kieranricardo committed:
full run with land detection

kieranricardo committed:
full run with land detection

Commit:
updated script

kieranricardo committed:
imroved vapour initialisation

kieranricardo committed:
Merge remote-tracking branch 'origin/intensity' into intensity

kieranricardo committed:
fix 0 storms

kieranricardo committed:
working run

kieranricardo committed:
working run

kieranricardo committed:
ocean profile climatology

Commit:
ocean clim script

Commit:
climatology, wind extraction

Commit:
edits

kieranricardo committed:
edits

wcarthur commented:
@kieranricardo did you make any mods to `hurr.f` from the original source? If so, just add a comment at the top of that file noting any changes.

kieranricardo committed:
fixed scaling bug

kieranricardo committed:
added documentation

kieranricardo committed:
edits

kieranricardo committed:
edits

kieranricardo committed:
edits

kieranricardo commented:
hey @wcarthur - I've added some documentation and fixed that bug I was talking about. Can have a look at this code and see whether it makes sense to you how to run/use it?

here's still the issue of too many low intensity storms though, this might be a parameter issue. Also Emanuel tends to initialise storms with 20m/s vortices rather than initial vortices sampled from the historical record so maybe a weakness of this model is that the threshold of intensity required for a vortex to intensify is unrealistically high. Anyway I'll pop this into a word doc

kieranricardo committed:
fixed ocean mixing scaling

wcarthur commented on intensity/temp_extract.sh:
```diff
@@ -0,0 +1,21 @@
+module purge
+
+module use /g/data/v10/public/modules/modulefiles
+module load dea/20210527
```
is this interchangeable with 

```
module use /g/data/hh5/public/modules
module load conda/analysis3
```

kieranricardo replied:
I'm not in hh5 but I only us:

- xarray
- gdal
- numpy
- pyproj
- scipy
- mpi4py

wcarthur commented on intensity/test_intensity.py:
```diff
@@ -0,0 +1,42 @@
+import os
+import sys
+sys.path.append(os.path.expanduser("~/geoscience/repos/tcha"))
```
Prefer to append to $PYTHONPATH in the shell rather than in the python code script

wcarthur reviewed (changes requested):
@kieranricardo Can you confirm if there's any input data in `~/geoscience/data`  

Can you check if the python modules available through `v10` match with what's available through `hh5`? THese are used in the `temp_extract.sh` scripts. I've swapped the `v10` `dea20210527` with `hh5`'s `conda/analysis3` python environment in other code with no problem

kieranricardo commented:
@wcarthur all the data is now in /g/data/w85/tc_intensity_data

kieranricardo committed:
review comments

kieranricardo committed:
review coments

kieranricardo committed:
review coments

kieranricardo committed:
review coments

kieranricardo committed:
notebook to analyze predicted and simulated landfall intensity

kieranricardo committed:
notes about required modules
